bentoml/cli/utils.py

```diff
@@ -92,6 +93,43 @@ def parse_key_value_pairs(key_value_pairs_str):
     return result
 
 
+def _echo_docker_api_result(docker_generator):
+    layers = {}
+    for line in docker_generator:
+        if "stream" in line:
+            cleaned = line['stream'].rstrip("\n")
+            if cleaned != "":
+                _echo(cleaned)
+        if "status" in line and line["status"] == "Pushing":
+            progress = line["progressDetail"]
+            layers[line["id"]] = progress["current"], progress["total"]
```

Comment:
When you get multiple lines with the same `"id"` key, does `progress["total"]` ever change?

Reply:
No, but each 'loop' may not have more layers than the previous so we need to recompute the sum anyways. There could technically be some caching done here, but I don't think the performance gain is worth the sacrifice in readability